source/common/http/conn_manager_utility.cc

```diff
@@ -100,6 +100,18 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
   request_headers.removeProxyConnection();
   request_headers.removeTransferEncoding();
 
+  // Sanitize referer field if exists.
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.sanitize_http_header_referer")) {
+    auto result = request_headers.get(Http::CustomHeaders::get().Referer);
+    if (!result.empty()) {
+      Utility::Url url;
+      if (result.size() > 1 || !url.initialize(result[0]->value().getStringView(), false)) {
+        // A request header shouldn't have multiple referer field.
+        request_headers.remove(Http::CustomHeaders::get().Referer);
```

Comment:
I wonder if we should handle multiple values if they're the same.  if the spec disallows I guess start simple and see if anyone complains :-P

Reply:
Yeah the spec doesn't specify on this. We'll see if people complain.

Reply:
Yes, the spec has always been clear on this; it is not a list header and its definition is absolute; 
https://datatracker.ietf.org/doc/html/rfc7231#section-5.5.2

     Referer = absolute-URI / partial-URI